processor/tailsamplingprocessor/internal/sampling/ottl.go

```diff
@@ -0,0 +1,126 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sampling // import "github.com/open-telemetry/opentelemetry-collector-contrib/processor/tailsamplingprocessor/internal/sampling"
+
+import (
+	"context"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/filter/expr"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/internal/filter/filterottl"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottlspan"
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl/contexts/ottlspanevent"
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/pdata/pcommon"
+	"go.uber.org/zap"
+)
+
+type ottlConditionFilter struct {
+	sampleSpanExpr      expr.BoolExpr[ottlspan.TransformContext]
+	sampleSpanEventExpr expr.BoolExpr[ottlspanevent.TransformContext]
+	ErrorMode           ottl.ErrorMode
+	logger              *zap.Logger
+}
+
+var _ PolicyEvaluator = (*ottlConditionFilter)(nil)
+
+// NewOTTLConditionFilter looks at the trace data and returns a corresponding SamplingDecision.
+func NewOTTLConditionFilter(logger *zap.Logger, spanConditions, spanEventConditions []string, errMode ottl.ErrorMode) (PolicyEvaluator, error) {
+	filter := &ottlConditionFilter{
+		ErrorMode: errMode,
+		logger:    logger,
+	}
+
+	set := component.TelemetrySettings{
```

Comment:
This is better than before, but still not as good as using the component's instance to the telemetry settings.  Currently only the logger is used, but it is always possible OTTL will want something else in the struct in the future.  I believe the change would be to pass `component.TelemetrySettings` from `createTracesProcessor` -> `newTracesProcessor` -> `getPolicyEvaluator` -> `getSharedPolicyEvaluator`.  All the other NewPolicy functions would still get the Logger from that struct, which is how it currently works.

Reply:
OK, thanks for your timely review! so most functions accept a `logger`, which we've switched to a `settings`. Please review it again.